web-bundle/src/main/java/com/graphhopper/resources/MVTResource.java

```diff
@@ -0,0 +1,149 @@
+package com.graphhopper.resources;
+
+import com.graphhopper.GraphHopper;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.util.DefaultEdgeFilter;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.storage.NodeAccess;
+import com.graphhopper.storage.index.LocationIndexTree;
+import com.graphhopper.util.EdgeExplorer;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.PointList;
+import com.graphhopper.util.StopWatch;
+import com.graphhopper.util.shapes.BBox;
+import com.wdtinc.mapbox_vector_tile.VectorTile;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.IGeometryFilter;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.JtsAdapter;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.TileGeomResult;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.UserDataKeyValueMapConverter;
+import com.wdtinc.mapbox_vector_tile.build.MvtLayerBuild;
+import com.wdtinc.mapbox_vector_tile.build.MvtLayerParams;
+import com.wdtinc.mapbox_vector_tile.build.MvtLayerProps;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.LineString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Path("mvt")
+public class MVTResource {
+
+    private static final Logger logger = LoggerFactory.getLogger(MVTResource.class);
+    private static final MediaType PBF = new MediaType("application", "x-protobuf");
+    private final GraphHopper graphHopper;
+    private final EncodingManager encodingManager;
+
+    @Inject
+    public MVTResource(GraphHopper graphHopper, EncodingManager encodingManager) {
+        this.graphHopper = graphHopper;
+        this.encodingManager = encodingManager;
+    }
+
+    @GET
+    @Path("{z}/{x}/{y}.mvt")
+    @Produces("application/x-protobuf")
+    public Response doGetXyz(
+            @Context HttpServletRequest httpReq,
+            @Context UriInfo uriInfo,
+            @QueryParam("vehicle") @DefaultValue("car") String vehicle,
+            @PathParam("z") int zInfo,
+            @PathParam("x") int xInfo,
+            @PathParam("y") int yInfo) {
+
+        if (zInfo <= 9) {
+            VectorTile.Tile.Builder mvtBuilder = VectorTile.Tile.newBuilder();
+            return Response.fromResponse(Response.ok(mvtBuilder.build().toByteArray(), PBF).build())
```

Comment:
With maputnik this gives me `Error: http status 200 returned without content`. Can we do better at returning 'an empty tile' ?

Reply:
I think it should be possible to convert an empty geometry to an MVT tile.

Reply:
As I cannot reproduce this problem (and also do not know how to create an empty geometry), can we delay and fix this in a later issue?